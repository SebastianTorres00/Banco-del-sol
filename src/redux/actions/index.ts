import axios from 'axios';
import firebase from 'firebase';
import { resFromBack, userType } from '../../types/Types';

require('firebase/firebase-auth');

export const REGISTER = 'REGISTER';
export const SET_USER = 'SET_USER';
export const SET_ACCOUNT = 'SET_ACCOUNT';
export const LOG_OUT = 'LOG_OUT';
export const FETCH_CVU = 'FETCH_CVU';
export const SET_TOKEN = 'SET_TOKEN';
export const GET_EMAIL = 'GET_EMAIL';
export const GET_DETAILS = 'GET_DETAILS';
export const GET_NAME = 'GET_NAME';

export function register(user: userType, password: string) {
  console.log('user, password', user, password);

  return (dispatch: any) => {
    firebase
      .auth()
      .createUserWithEmailAndPassword(user.email.toLowerCase(), password)
      .then(response => {
        console.log('response', response);
        console.log('process.env.IP_ADDRESS', process.env.IP_ADDRESS);
        response.user
          ?.getIdToken()
          .then(idToken => {
            console.log('idtoken', idToken);
            axios
              .post<resFromBack>(`http://localhost:3001/api/user/`, user, {
                headers: {
                  authorization: `Bearer ${idToken}`,
                },
              })
              .then(responseAgain => {
                console.log('responseAgain', responseAgain);
                dispatch({
                  type: SET_ACCOUNT,
                  payload: responseAgain.data.account,
                });
                dispatch({
                  type: SET_USER,
                  payload: responseAgain.data.user,
                });
                dispatch({
                  type: SET_TOKEN,
                  payload: idToken,
                });
              });
          })
          .catch(error => console.error(error, 'error'));
      })
      .catch(error => console.error(error, 'error'));
  };
}
export function login(email: string, password: string) {
  return (dispatch: any) => {
    firebase
      .auth()
      .signInWithEmailAndPassword(email.toLowerCase(), password)
      .then(response => {
        response.user
          ?.getIdToken()
          .then(idToken => {
            axios
              .get<resFromBack>(
                `http://192.168.0.4:3001/api/user/?email=${email}`,
                {
                  headers: {
                    authorization: `Bearer ${idToken}`,
                  },
                },
              )
              .then(responseFromBack => {
                dispatch({
                  type: SET_ACCOUNT,
                  payload: responseFromBack.data.account,
                });
                dispatch({
                  type: SET_USER,
                  payload: responseFromBack.data.user,
                });
                dispatch({
                  type: SET_TOKEN,
                  payload: idToken,
                });
              });
          })
          .catch(error => console.error(error));
      })
      .catch(error => console.error(error));
  };
}

export function logout() {
  return (dispatch: any) => {
    firebase
      .auth()
      .signOut()
      .then(() => {
        dispatch({
          type: LOG_OUT,
          payload: {},
        });
      });
  };
}

export function fetchCvu() {
  return (dispatch: any) => {
    dispatch({
      type: FETCH_CVU,
      payload: true,
    });
  };
}

export function addFunds(
  senderEmail: string,
  receiverEmail: string,
  type: string,
  value: number,
  comment: string,
  token: string,
  dispatch: any,
) {
  axios
    .post(
      `http://localhost:3001/api/account`,
      { senderEmail, receiverEmail, type, value, comment },
      {
        headers: {
          authorization: `Bearer ${token}`,
        },
      },
    )
    .then(response => {
      dispatch({
        type: SET_ACCOUNT,
        payload: response.data,
      });
    })
    .catch(error => {
      alert('El usuario no existe');
      console.log(error);
    });
}

export const getEmail =
  (emailUser: string, idToken: string, nameUser: string) => dispatch => {
    console.log(emailUser, 'ACTIONS');

    axios
      .get(`http://192.168.0.4:3001/api/contacts/${emailUser}`, {
        headers: {
          authorization: `Bearer ${idToken}`,
        },
      })
      .then(details => {
        console.log('details', details.data);
        const { email, cvu } = details.data;

        dispatch({
          type: GET_EMAIL,
          payload: { name: nameUser, email, cvu },
        });

        dispatch({
          type: GET_NAME,
          payload: ``,
        });
      })
      .catch(error => {
        return alert(
          'En este momento no se encuentra disponible la posibilidad de agregar a personas a tu lista.',
        );
      });
  };

export const getName = (emailUser: string, idToken: string) => dispatch => {
  axios
    .get(`http://192.168.0.4:3001/api/contacts/${emailUser}`, {
      headers: {
        authorization: `Bearer ${idToken}`,
      },
    })
    .then(details => {
      const { name, lastName } = details.data;
      dispatch({
        type: GET_NAME,
        payload: `${name} ${lastName}`,
      });
    })

    .catch(error => {
      dispatch({
        type: GET_NAME,
        payload: ``,
      });
      return alert(
        'Este usuario no se encuentra registrado, proximamente lo podras invitar!',
      );
    });
};

export const detailContact = (email: string, name: string) => dispatch => {
  dispatch({
    type: GET_DETAILS,
    payload: { name, email },
  });
};
export async function updateAccount(
  email: string,
  token: string,
  dispatch: any,
) {
  axios
    .get(`http://192.168.0.4:3001/api/account/?email=${email}`, {
      headers: {
        authorization: `Bearer ${token}`,
      },
    })
    .then(response => {
      dispatch({
        type: SET_ACCOUNT,
        payload: response.data,
      });
    })
    .catch(error => console.log(error));
}
