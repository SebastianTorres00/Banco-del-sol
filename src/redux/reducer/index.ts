import {
  REGISTER,
  SET_USER,
  SET_ACCOUNT,
  LOG_OUT,
  FETCH_CVU,
  SET_TOKEN,
  GET_EMAIL,
  GET_DETAILS,
  GET_NAME,
  SET_ERROR,
} from '../actions/index';

interface actionType {
  type: string;
  payload: Object;
}

const initialState = {
  account: {},
  user: {},
  token: '',
  cvuAsociate: false,
  Contacts: [],
  DetailTransfer: {},
  nameDetail: '',
  errors: {},
};

export default function rootReducer(state = initialState, action: actionType) {
  switch (action.type) {
    case REGISTER:
      return {
        ...state,
        user: action.payload,
      };
    case SET_USER:
      return {
        ...state,
        user: action.payload,
      };
    case SET_ACCOUNT:
      return {
        ...state,
        account: action.payload,
      };
    case LOG_OUT:
      return {
        ...state,
        account: action.payload,
        user: action.payload,
      };
    case FETCH_CVU:
      return {
        ...state,
        cvuAsociate: action.payload,
      };
    case SET_TOKEN:
      return {
        ...state,
        token: action.payload,
      };
    case GET_EMAIL:
      return {
        ...state,
        Contacts: [
          ...state.Contacts,
          {
            name: action.payload.name,
            lastName: action.payload.lastName,
            email: action.payload.email,
            cvu: action.payload.cvu,
          },
        ],
      };
    case GET_DETAILS:
      return {
        ...state,
        DetailTransfer: {
          name: action.payload.name,
          email: action.payload.email,
        },
      };
    case GET_NAME:
      return {
        ...state,
        nameDetail: action.payload,
      };
    default:
      return state;
  }
}
