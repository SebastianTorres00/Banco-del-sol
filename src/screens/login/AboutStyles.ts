import { StyleSheet } from 'react-native';
import colors from '../../constants/colors';

export const stylesAbout = StyleSheet.create({
  about: {
    zIndex: 10,
    backgroundColor: '#fff',
    width: 342,
    maxHeight: 400,
    minHeight: 400,
    position: 'absolute',
    marginLeft: '6%',
    alignItems: 'center',

    marginTop: '-10%',

    borderRadius: 15,

    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 6,
    },
    shadowOpacity: 0.37,
    shadowRadius: 7.49,

    elevation: 12,
  },
  btnAbout: {
    width: 250,
    alignItems: 'center',
    marginTop: 20,
    marginBottom: 20,
    borderRadius: 25,
    borderWidth: 1,
    borderColor: colors.primary,
    backgroundColor: '#ffffff',
    padding: 15,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 7,
    },
    shadowOpacity: 0.25,
    shadowRadius: 26,
    elevation: 14,
  },
  btnText: {
    color: colors.primary,
    paddingLeft: 5,
    paddingRight: 5,
    alignSelf: 'center',
    fontSize: 15,
    fontWeight: '700',
    marginBottom: 8,
  },
  tinyLogo: {
    width: 300,
    height: 150,
  },
  // Abot two
  tinyLogoTwo: {
    width: 300,
    height: 150,
  },
  aboutSelectPhone: {
    right: 0,
    zIndex: 10,
    backgroundColor: '#fff',
    width: 300,
    maxHeight: 185,
    minHeight: 185,
    position: 'absolute',
    marginRight: '8%',
    alignItems: 'center',
    alignSelf: 'center',
    marginTop: '-10%',

    borderRadius: 15,

    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 6,
    },
    shadowOpacity: 0.37,
    shadowRadius: 7.49,

    elevation: 12,
  },
  btnTextSelect: {
    color: colors.primary,
    paddingLeft: 5,
    paddingRight: 5,
    alignSelf: 'center',
    fontSize: 18,
    fontWeight: '700',
    marginTop: '15%',

    textAlign: 'center',
  },

  btnAboutSelect: {
    height: 30,
    width: 115,
    justifyContent: 'space-evenly',

    alignSelf: 'center',
    alignItems: 'center',

    marginLeft: 10,

    borderRadius: 25,
    borderWidth: 1,
    borderColor: colors.primary,
    backgroundColor: '#ffffff',
    padding: 15,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 7,
    },
    shadowOpacity: 0.25,
    shadowRadius: 26,
    elevation: 14,
  },
});
