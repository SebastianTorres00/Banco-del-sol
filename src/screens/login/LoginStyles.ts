import { StyleSheet } from 'react-native';
import colors from '../../constants/colors';

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.white,
    justifyContent: 'center',
    overflow: 'hidden',
    zIndex: 0,
    alignSelf: 'center',
    minHeight: 785,
    maxHeight: 785,
    width: 388,
    marginLeft: -2,
    borderRadius: 30,
  },
  inputContainer: {
    flex: 1,
    backgroundColor: colors.white,
    justifyContent: 'center',
  },
  link: {
    marginTop: 10,
    marginBottom: 10,
    color: 'orange',
    marginLeft: 'auto',
    marginRight: 'auto',
    fontSize: 15,
    fontWeight: 'bold',
  },
  ellipse: {
    position: 'relative',
    zIndex: 20,
    bottom: '-25%',
    marginTop: '-50%',
    left: '20%',
    width: '60%',
    height: '60%',
    borderRadius: 999,
    transform: [{ scaleX: 3 }],
  },
  button: {
    zIndex: 80,
    width: '90%',
    alignItems: 'center',
    marginTop: 20,
    marginBottom: 20,
    borderRadius: 25,
    borderWidth: 2,
    borderColor: colors.primary,
    backgroundColor: '#ffffff',
    padding: 15,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 7,
    },
    shadowOpacity: 0.25,
    shadowRadius: 26,
    elevation: 14,

    marginLeft: 'auto',
    marginRight: 'auto',
  },
  buttondisabled: {
    zIndex: 80,
    width: '90%',
    alignItems: 'center',
    marginTop: 20,
    marginBottom: 20,
    borderRadius: 25,
    borderWidth: 2,
    borderColor: colors.primary,
    backgroundColor: '#deac98de',
    padding: 15,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 7,
    },
    shadowOpacity: 0.25,
    shadowRadius: 26,
    elevation: 14,

    marginLeft: 'auto',
    marginRight: 'auto',
  },
  text: {
    color: colors.primary,
    paddingLeft: 5,
    paddingRight: 5,
    alignSelf: 'center',
    fontSize: 28,
    fontWeight: 'bold',
  },
  textlogin: {
    color: '#9a788ccf',
    paddingLeft: 5,
    paddingRight: 5,
    alignSelf: 'center',
    fontSize: 28,
    fontWeight: 'bold',
  },
  btns: {
    width: 300,
    height: 100,
    zIndex: 30,
    marginBottom: -150,
    marginLeft: 'auto',
    marginRight: 'auto',
  },
  header: {
    height: 75,
    width: '100%',
  },

  title: {
    height: 70,
    justifyContent: 'center',
    alignItems: 'center',
    position: 'absolute',
    left: 70,
    top: 0,
    marginTop: '-0.3%',
    width: '30%',
  },

  textTitle: {
    position: 'absolute',
    left: 5,
    top: 15,
    fontSize: 20,
    color: 'white',
    fontWeight: '500',
    width: '180%',
  },

  back: {
    position: 'absolute',
    height: 70,
    width: 70,
    left: 0,
    top: 0,
    justifyContent: 'center',
    alignItems: 'center',
  },

  headerOne: { backgroundColor: '#000', height: 63 },

  icon: { marginTop: '-18%' },

  // ABOUT
  // 628 / 510
  about: {
    zIndex: 10,
    // backgroundColor: 'red',
    width: '33%',
    // height: '65%',
    maxHeight: 738,
    minHeight: 738,
    position: 'absolute',
    marginLeft: '4%',
    justifyContent: 'center',

    marginTop: '-10%',
  },
  aboutTwo: {
    width: '33%',
    // height: '65%',
    maxHeight: 828,
    // minHeight: 828,
    position: 'absolute',
    // marginLeft: '-4%',
    marginRight: '4%',
    justifyContent: 'center',

    marginTop: '-10%',
    right: 0,
    // backgroundColor: 'red',
  },
  iconLinkedin: {
    borderRadius: 50,
  },

  iconGit: {
    borderRadius: 50,
  },

  imgBx: {
    width: '100%',
    height: '100%',

    backgroundColor: '#fff',
    marginRight: '4%',

    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 5,
    },
    shadowOpacity: 0.34,
    shadowRadius: 6.27,

    elevation: 10,

    transform: [{ rotate: '3deg' }],

    marginTop: '5%',
  },

  imgBoxTwo: {
    // width: 289,
    // height: 315,

    width: '48%',
    height: '65%',

    backgroundColor: '#fff',
    // marginRight: '4%',

    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 5,
    },
    shadowOpacity: 0.34,
    shadowRadius: 6.27,

    elevation: 10,

    transform: [{ rotate: '-3deg' }],

    marginTop: '7%',
  },

  textAbout: {
    color: '#fff',
    fontSize: 30,
    fontWeight: '600',
    textAlign: 'center',
  },

  imgProfile: {
    width: '30%',
    borderRadius: 50,
    alignSelf: 'center',
    marginTop: '15%',

    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 10,
    },
    shadowOpacity: 0.53,
    shadowRadius: 13.97,

    elevation: 21,
  },
});
