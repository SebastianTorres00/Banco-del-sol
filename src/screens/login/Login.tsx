import React, { useEffect, useState } from 'react';
import { View, TouchableOpacity, Text, TextInput, Image } from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import { LinearGradient } from 'expo-linear-gradient';
import { AntDesign } from '@expo/vector-icons';
import AwesomeAlert from 'react-native-awesome-alerts';
import {
  cleanErrors,
  login,
  setLoadingFalse,
  setLoadingTrue,
} from '../../redux/actions';
import { styles } from './LoginStyles';
import { stylesAbout } from './AboutStyles';
import { Props, resFromBack, RootState } from '../../types/Types';
import { ButtonPrimaryStyle } from '../../constants/ButtonPrymaryStyle';
import colors from '../../constants/colors';
import { LoadingFull } from '../loading2/LoadingFull';
import { StylesCon } from '../../constants/Styles';

export const Login = ({ navigation }: Props) => {
  const userStore = useSelector((state: resFromBack) => state.user);
  const loading = useSelector((state: RootState) => state.loading);
  const error = useSelector((state: RootState) => state.errors);
  const dispatch = useDispatch();
  const [state, setState] = useState(false);
  const [num, setNum] = useState(100);
  const [numTest, setNumTest] = useState(0);
  const [user, setUser] = useState({
    email: 'ferro@hotmail.com',
    password: '123QWE&',
    amount: 0,
  });
  useEffect(() => {
    if (error.length) {
      setState(true);
    }
  }, [error.length]);

  useEffect(() => {
    // Esto es provicional
    if (numTest === 0) {
      document.body.style.zoom = `${num}%`;
    } else {
      document.body.style.zoom = `${num}%`;
    }
  }, [num]);

  if (error.length) {
    dispatch(setLoadingFalse());
  }

  function Zoom() {
    setNum(num + 10);
    setTimeout(() => {
      setNumTest(0);
    }, 500);
  }

  function ZoomOut() {
    setNum(num - 10);
    setTimeout(() => {
      setNumTest(1);
    }, 1000);
  }

  return (
    <View style={StylesCon.phone}>
      <img
        style={{
          width: '100%',
          height: '100%',
          position: 'absolute',
        }}
        src="https://cdn.discordapp.com/attachments/872492726397042688/880917630180028436/Banco-del-Sol-Background_Web.png"
        alt=""
      />
      <View style={stylesAbout.about}>
        <TouchableOpacity
          style={stylesAbout.btnAbout}
          onPress={() => navigation.push('AboutUs')}
        >
          <Text style={stylesAbout.btnText}>Sobre Nosotros</Text>
        </TouchableOpacity>

        <TouchableOpacity style={stylesAbout.btnAbout}>
          <Text style={stylesAbout.btnText}>Administrador</Text>
        </TouchableOpacity>

        <View style={{ width: '100%', alignItems: 'center', bottom: 0 }}>
          <Image
            style={stylesAbout.tinyLogo}
            source={{
              uri: 'https://media.discordapp.net/attachments/872492726397042688/874643755158892594/Banco-del-Sol-Logo.png',
            }}
          />
        </View>
      </View>

      <img
        style={{
          width: 411,
          height: 813,
          position: 'absolute',
          alignSelf: 'center',
          marginLeft: '1%',
          marginRight: '1%',
        }}
        src="https://cdn.discordapp.com/attachments/872492726397042688/880174589605478400/Mockup_-_android_-_BDS_app.png"
        alt=""
      />

      <View style={StylesCon.filler} />
      <View style={styles.container}>
        <View style={StylesCon.headerOne}>
          <LinearGradient
            style={StylesCon.header}
            colors={[colors.primary, colors.secondary]}
          />
          <View style={StylesCon.title}>
            <Text style={styles.textTitle}>Inicio</Text>
          </View>
          <TouchableOpacity
            style={StylesCon.back}
            onPress={() => {
              navigation.popToTop();
            }}
          >
            <AntDesign
              name="arrowleft"
              size={35}
              color="white"
              style={StylesCon.icon}
            />
          </TouchableOpacity>
        </View>
        <LoadingFull show={loading} />
        <View
          style={{
            alignSelf: 'center',
            zIndex: 100,
            position: 'absolute',
            width: '100%',
          }}
        >
          <AwesomeAlert
            show={state}
            showProgress={false}
            title={error}
            closeOnTouchOutside={false}
            closeOnHardwareBackPress={false}
            showCancelButton={false}
            showConfirmButton
            confirmText="Aceptar"
            confirmButtonColor="#ff4b6e"
            onConfirmPressed={() => {
              setState(false);
              dispatch(cleanErrors());
            }}
          />
        </View>
        <View style={styles.inputContainer}>
          <TextInput
            placeholder="Email...."
            placeholderTextColor="grey"
            value={user.email}
            onChangeText={(text: string) => setUser({ ...user, email: text })}
            keyboardType="email-address"
            style={ButtonPrimaryStyle.input}
          />

          <TextInput
            placeholder="Contraseña..."
            placeholderTextColor="grey"
            value={user.password}
            onChangeText={(text: string) =>
              setUser({ ...user, password: text })
            }
            secureTextEntry
            style={ButtonPrimaryStyle.input}
          />
          <TouchableOpacity
            style={styles.link}
            onPress={() => navigation.push('ForgotPassword')}
          >
            <Text style={styles.link}>¿Olvidaste tu contraseña?</Text>
          </TouchableOpacity>
        </View>

        <View style={styles.btns}>
          {user.password.length > 6 && user.email.length > 4 ? (
            <TouchableOpacity
              onPress={() => {
                dispatch(login(user.email, user.password));
                dispatch(setLoadingTrue());
              }}
              style={styles.button}
            >
              <Text style={styles.text}>INGRESAR</Text>
            </TouchableOpacity>
          ) : (
            <TouchableOpacity
              onPress={() => {
                console.log();
              }}
              disabled
              style={styles.buttondisabled}
            >
              <Text style={styles.textlogin}>INGRESAR</Text>
            </TouchableOpacity>
          )}
        </View>
        <LinearGradient
          style={styles.ellipse}
          colors={[colors.primary, colors.secondary]}
          end={[1, 1]}
        />
        {userStore.email &&
        userStore.email.length &&
        userStore.condition === 'active'
          ? navigation.push('HomeTab')
          : null}
      </View>
      <View style={stylesAbout.aboutSelectPhone}>
        <Text style={stylesAbout.btnTextSelect}>
          Elige tu resolucion de pantalla
        </Text>
        <View
          style={{
            position: 'absolute',
            width: '36%',
            bottom: 100,
          }}
        >
          <TouchableOpacity
            style={stylesAbout.btnAboutSelect}
            onPress={() => ZoomOut()}
          >
            <Text style={stylesAbout.btnText}>MENOS</Text>
          </TouchableOpacity>
          <TouchableOpacity
            style={stylesAbout.btnAboutSelect}
            onPress={() => Zoom()}
          >
            <Text style={stylesAbout.btnText}>MAS</Text>
          </TouchableOpacity>
        </View>
      </View>
      <View style={StylesCon.filler} />
    </View>
  );
};
