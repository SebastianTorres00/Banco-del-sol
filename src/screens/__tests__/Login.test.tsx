/* @jest-environment jsdom */

import React from 'react';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { initialState } from '../../redux/reducer';
import { Login } from '../login/Login';

const middlewares: any[] = [];
const mockStore = configureStore(middlewares);

describe('Login', () => {
  it('renders the login', () => {
    const push = jest.fn();
    const mockNavigation: any = { push };
    const store = mockStore(initialState);
    const wrapper = mount(
      <Provider store={store}>
        <Login navigation={mockNavigation} />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
