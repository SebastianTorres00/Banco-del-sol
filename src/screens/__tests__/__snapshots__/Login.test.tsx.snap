// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders the component 1`] = `ReactWrapper {}`;
